dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/tcp/LibertyLoggingHandler.java

```diff
@@ -0,0 +1,161 @@
+	}
+
+	@Override
+	public void channelActive(ChannelHandlerContext ctx) throws Exception {
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+			Tr.event(ctx.channel(), tc, "SocketChannel accepted, local: " + ctx.channel().localAddress() + " remote: " + ctx.channel().remoteAddress());
+		}
+		ctx.fireChannelActive();
+	}
+
+
+	@Override
+	public void bind(ChannelHandlerContext ctx, SocketAddress localAddress, ChannelPromise promise) throws Exception {
+		// Used when binding a Channel to a port to accept connections
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+			Tr.event(ctx.channel(), tc, "Calling bind, local: " + localAddress);
+		}
+		super.bind(ctx, localAddress, promise);
+	}
+
+	@Override
+	public void connect(
+			ChannelHandlerContext ctx,
+			SocketAddress remoteAddress, SocketAddress localAddress, ChannelPromise promise) throws Exception {
+		// Used when connecting a channel to an endpoint
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+			Tr.event(ctx.channel(), tc, "Calling connect, local: " +localAddress + " remote: " + remoteAddress);
+		}
+		super.connect(ctx, remoteAddress, localAddress, promise);
+	}
+
+
+	@Override
+	public void close(ChannelHandlerContext ctx, ChannelPromise promise) throws Exception {
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+			Tr.event(ctx.channel(), tc, "SocketChannel closing, local: " + ctx.channel().localAddress() + " remote: " + ctx.channel().remoteAddress());
+		}
+		ctx.close(promise);
+	}
+
+	@Override
+	public void disconnect(ChannelHandlerContext ctx, ChannelPromise promise) throws Exception {
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+			Tr.event(ctx.channel(), tc, "Channel disconnected, local: " + ctx.channel().localAddress() + " remote: " + ctx.channel().remoteAddress());
+		}
+		ctx.disconnect(promise);
+	}
+
+	@Override
+	public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+			Tr.event(ctx.channel(), tc, "read completed. Read: " + ctx.channel().attr(BYTES_READ_KEY).getAndSet(0) + " bytes");
+		}
+		ctx.fireChannelReadComplete();
+	}
+
+	@Override
+	public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+			Tr.event(ctx.channel(), tc, "read (async) called for local: " + ctx.channel().localAddress() + " remote: " + ctx.channel().remoteAddress()+ (msg instanceof ByteBuf ? " bytes read: "+((ByteBuf) msg).readableBytes() : " object read: " + msg));
+		}
+		if(msg instanceof ByteBuf) {
+			ctx.channel().attr(BYTES_READ_KEY).getAndSet(ctx.channel().attr(BYTES_READ_KEY).get() + ((ByteBuf) msg).readableBytes());
+		}else {
+			if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+				Tr.debug(ctx.channel(), tc, "Could not get bytes for object: " + msg);
```

Comment:
Are there any  plans to support non ByteBuf object reads/write? Should we consider this an error case/unsupported scenario? I understand that the position of the handler in the pipeline determines the typing that can be passed on, but I'd imagine that we would aim to position this handler in a consistent fashion and passing along the ByteBuf used to read/write data.

Reply:
I'll adapt the class access modifier so that it should only be used inside the package itself. This should help ensure that we don't hit this issue so I'll also adapt the code to log as if we only expect ByteBuf objects. Thinking about this in a more broader sense, we should discuss ways to ensure that the pipeline is not modified outside the bundle which could affect things like timeouts and logging. Created https://github.com/OpenLiberty/open-liberty/issues/23435 to follow up on this